src/includes/classes/AdvCacheBackCompat.php

```diff
@@ -66,4 +66,23 @@ public static function zenCacheConstants()
 
         unset($_constant, $_value, $_global_ns); // Housekeeping.
     }
+
+    /**
+     * Back compat. with `COMET_CACHE_ALLOW_BROWSER_CACHE` constants.
+     *
+     * @since 16xxxx Renaming COMET_CACHE_ALLOW_BROWSER_CACHE to COMET_CACHE_ALLOW_CLIENT_SIDE_CACHE
+     */
+    public static function browserCacheConstant()
+    {
+        $_global_ns = mb_strtoupper(GLOBAL_NS);
+
+        if (!($constants = get_defined_constants(true)) || empty($constants['user'])) {
```

Comment:
This looks great to me. However, you can optimize this further by ditching the section that collects user-defined constants; i.e., get rid of `get_defined_constants()` and instead of the `in_array()` check, just use `if (defined('COMET_CACHE_ALLOW_BROWSER_CACHE')) {`

The reason is because this back compat. routine is only looking for one _specific_ constant and not any that have a special prefix like the other method does. So it's easier, and simpler will be equal to faster.